stl/inc/format

```diff
@@ -343,13 +347,20 @@ public:
     // _Next_arg_id > 0 means automatic
     // _Next_arg_id == -1 means manual
     constexpr size_t next_arg_id() {
-        if (_Next_arg_id >= 0) {
-            return _Next_arg_id++;
+        if (_Next_arg_id == -1) {
+            throw format_error("Can not switch from manual to automatic indexing");
         }
-        throw format_error("Can not switch from manual to automatic indexing");
+        return static_cast<size_t>(_Next_arg_id++);
     }
-    constexpr void check_arg_id(size_t _Id) {
-        (void) _Id;
+    constexpr void check_arg_id(const size_t _Id) {
+        if (_STD is_constant_evaluated()) {
+            if (_Id >= _Num_args) {
+                _TRY_BEGIN
+                throw 42;

```

Comment:
Should this be commented? It seems mysterious if a user were investigating a `constexpr` evaluation failure.

Reply:
Added a comment